keyboards/draytronics/daisy_v2/keymaps/default/keymap.c

```diff
@@ -0,0 +1,118 @@
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+This code is inspired by and adapted from the code used in the Printed Pad by Noah Beidelman (@noahbei)
+
+It also references the concept of glitching animations from Aleks (@aleksbrgt)
+
+The pixel graphics used here are from a combination of sources;
+
+1. Layer indicators are created by myself and free to use by anyone.
+2. "Revengeday", "Cyber Cafe", "Cortex Implant" logos are used with kind permission of OBEY THE SYSTEM.
+   A collective of Fediverse instances and creatives. https://git.cyberwa.re/obey-the-system.
+   They are licenced as Non-Commercial and for use by members of the network, with attribution.
+3. Key press indicator graphics were commissioned for this project and were designed by the
+   amazing https://corteximplant.com/@jadedtwin / https://www.jadedtwin.com/
+*/
+#include QMK_KEYBOARD_H
+
+enum layers {
+    _BASE,
+    _CODE
+};
+
+enum my_keycodes {
+  ENCODER_PRESS = QK_USER,
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+    /*
+     * ┌────────────┐
+     * │* See Note *│
+     * ├────────────┼────────────┬────────────┬─────────────┐
+     * │ Code Layer │ Media Next │ Media Prev │ Media Pause │
+     * ├────────────┼────────────┼────────────┼─────────────┤
+     * │ Prev Desk  │ Miss Ctrl  │ App Window │  Next Desk  │
+     * └────────────┴────────────┴────────────┴─────────────┘
+     */
+    [_BASE] = LAYOUT(
+        LT(0, ENCODER_PRESS),
+        MO(_CODE), KC_MPRV, KC_MNXT, KC_MPLY,
+        C(KC_LEFT), C(KC_UP), C(KC_DOWN), C(KC_RIGHT)
+    ),
+    /*
+     * ┌────────────┐
+     * │* See Note *│
+     * ├────────────┼────────────┬────────────┬─────────────┐
+     * │            │  RGB Mode  │   RBG Hue  │  RGB Toggle │
+     * ├────────────┼────────────┼────────────┼─────────────┤
+     * │ Scrn Shot  │ Force Quit │   GUI + F  │   DFU Mode  │
+     * └────────────┴────────────┴────────────┴─────────────┘
+     */
+    [_CODE] = LAYOUT(
+        LT(0, ENCODER_PRESS),
+        _______, RGB_MOD, RGB_HUI, RGB_TOG,
+        G(S(KC_5)), G(A(KC_ESC)), G(KC_F), QK_BOOT
+    )
+
+    /*
+    * See Note *
+    * Tapping on the rotary encoder switch will mute or unmute the volume.
+    * Pressing and holding will cycle through the OLED modes, they are as follows;
+    * 1. Layer indicator
+    * 2. Revengeday glitch logo
+    * 3. Cyber Cafe glitch logo
+    * 4. Cortex Implant glitch logo
+    * 5. Key press animation.
+    *
+    * By default the rotary encoder itself will adjust the volume but it can be adjusted by changing the encoder_update_user function.
+    */
+};
+
+#if defined(ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
+    [_BASE] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU)  },
+    [_CODE] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU)  },
+};
+#endif
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch(keycode) {
+        case LT(0, ENCODER_PRESS):
```

Comment:
As above. I wonder if you even need a custom keycode? Or if you would be able to just override the functionality of the existing `KC_MUTE` keycode, and use that in your keymap.
```suggestion
        case ENCODER_PRESS:
```

Reply:
I am open to other suggestions, I was not sure overriding keycodes allowed me to do complex action's (apologies if I am wrong) also is it not clearer to have a custom keycode (would not this be the point of them) for the end user to understand.